python/ray/tests/test_actor.py

```diff
@@ -1676,5 +1677,59 @@ def method(self):
     assert result == "ok"
 
 
+def test_get_actor_after_same_name_actor_dead(shutdown_only):
+    ray.init(namespace="test")
+
+    @ray.remote
+    class Actor:
+        def get_pid(self):
+            return os.getpid()
+
+    def get_pid(actor):
+        while True:
+            try:
+                return ray.get(actor.get_pid.remote())
+            except Exception:
+                pass
+
+    a = Actor.options(name="test", max_restarts=1).remote()
+
+    pid = get_pid(a)
```

Comment:
```suggestion
    pid = ray.get(a.get_pid.remote())
```
can just do this here and kill the helper function?